src/scripts/global-data.lua

```diff
@@ -54,24 +54,76 @@ local function build_amount_string(material)
-  -- temperature
-  local temperature = material.temperature
-  local temp_min = material.minimum_temperature
-  local temp_max = material.maximum_temperature
-  if temperature then
-    amount_string = amount_string.."  ("..math.round_to(temperature, 2).."°)"
-  elseif temp_min and temp_max then
-    if temp_min == -0X1.FFFFFFFFFFFFFP+1023 then
-      amount_string = amount_string.."  (<= "..math.round_to(temp_max, 2).."°)"
-    elseif temp_max == 0X1.FFFFFFFFFFFFFP+1023 then
-      amount_string = amount_string.."  (>= "..math.round_to(temp_min, 2).."°)"
-    else
-      amount_string = amount_string.."  ("..math.round_to(temp_min, 2).." - "..math.round_to(temp_max, 2).."°)"
+  -- second return is the "average" amount
+  return amount_string, amount == nil and ((material.amount_min + material.amount_max) / 2) or nil
+end
+
+local function build_temperature_strings(temperatureMin, temperatureMax)
+  if temperatureMin == temperatureMax then
+    return tostring(temperatureMin), " ("..tostring(temperatureMin).."°C"..")"
+  end
+
+  local min = -0X1.FFFFFFFFFFFFFP+1023
+  local max = 0X1.FFFFFFFFFFFFFP+1023
+
+  if temperatureMin > min and temperatureMax < max then 
+    return tostring(temperatureMin).."-"..tostring(temperatureMax), " ("..tostring(temperatureMin).."-"..tostring(temperatureMax).."°C"..")"
+  end
+
+  if temperatureMax < max then
+    return  "≤"..tostring(temperatureMax), " (≤"..tostring(temperatureMax).."°C"..")"
+  end
+
+  if temperatureMin > min then
+    return  "≥"..tostring(temperatureMin)," (≥"..tostring(temperatureMin).."°C"..")"
+  end
+end
+
+local function assureTemperatures(material, obj_data, fluid_data, recipe, secondRun)
+  local default = obj_data.default_temperature
+  local defaultMin = -0X1.FFFFFFFFFFFFFP+1023
+  local defaultMax = 0X1.FFFFFFFFFFFFFP+1023
+  
+  local isDefault
+  local temperatureMin
+  local temperatureMax
+
+  if material.temperature then
+    isDefault = material.temperature == default
+    temperatureMin = material.temperature
+    temperatureMax = material.temperature
+  elseif material.minimum_temperature or material.maximum_temperature then
+    temperatureMin = material.minimum_temperature
+    temperatureMax = material.maximum_temperature
+
+    if material.minimum_temperature <= defaultMin then
+      temperatureMin = defaultMin
     end
+
+    if material.maximum_temperature >= defaultMax then
+      temperatureMax = defaultMax
+    end
+
+    isDefault = false
+  else
+    isDefault = true
+    temperatureMin = default
+    temperatureMax = default
   end
 
-  -- second return is the "average" amount
-  return amount_string, amount == nil and ((material.amount_min + material.amount_max) / 2) or nil
+  local temperature_key, temperature_string = build_temperature_strings(temperatureMin, temperatureMax)
+
+  if not secondRun and not fluid_data.temperatures[temperature_key] and recipe.category ~= "py-venting" and recipe.category ~= "py-runoff" and recipe.category ~= "py-venting" then
```

Comment:
I'm not a fan of hardcoding recipe categories. What's the reason for these ones in particular?